SimConnectMSFS/SimConnectCache.cs

```diff
@@ -426,66 +449,135 @@ public void setEventID(string eventID)
             );
         }
 
-        public float GetSimVar(String SimVarName)
+        public FSUIPCOffsetType GetSimVar(String simVarName, out String stringVal, out double floatVal)
         {
-            float result = 0;
+            FSUIPCOffsetType simVarType = FSUIPCOffsetType.Float;
+            bool isFloat = false;
+            bool isString = false;
+
+            stringVal = "0";
+            floatVal = 0.0F;
+
             if (!IsConnected()) 
-                return result;
+                return simVarType;
 
-            if (!SimVars.Exists(lvar => lvar.Name == SimVarName))
+            isFloat = SimVars.Exists(lvar => lvar.Name == simVarName);
+            if(!isFloat)
             {
-                RegisterSimVar(SimVarName);            
-                WasmModuleClient.AddSimVar(m_oSimConnect, SimVarName, WasmRuntimeClientData);              
+                isString = StringSimVars.Exists(lvar => lvar.Name == simVarName);
+
+                if (!isString)
+                {
+                    simVarType = RegisterSimVar(simVarName);
+                    if (simVarType == FSUIPCOffsetType.String)
+                    {
+                        WasmModuleClient.AddStringSimVar(m_oSimConnect, simVarName, WasmRuntimeClientData);
+                    }
+                    else
+                    {
+                        WasmModuleClient.AddSimVar(m_oSimConnect, simVarName, WasmRuntimeClientData);
+                    }
+                }
+                else
+                {
+                    simVarType = FSUIPCOffsetType.String;
+                }
             }
 
-            result = SimVars.Find(lvar => lvar.Name == SimVarName).Data;
+            if(simVarType == FSUIPCOffsetType.Float)
+            {
+                floatVal = SimVars.Find(lvar => lvar.Name == simVarName).Data;
+            }
+            else
+            {
+                stringVal = StringSimVars.Find(lvar => lvar.Name == simVarName).Data;
+                stringVal = (stringVal) == null ? "0" : stringVal;
+            }
 
-            return result;
+            return simVarType;
         }
 
         public void SetSimVar(String SimVarCode)
         {
             WasmModuleClient.SetSimVar(m_oSimConnect, SimVarCode, WasmRuntimeClientData);   
         }
 
-        private void RegisterSimVar(string SimVarName)
+        private FSUIPCOffsetType RegisterSimVar(string SimVarName)
         {
-            SimVar NewSimVar = new SimVar() { Name = SimVarName, ID = (uint) (SimVars.Count + SIMVAR_DATA_DEFINITION_OFFSET) };
-            SimVars.Add(NewSimVar);
-
-            if (MaxClientDataDefinition >= NewSimVar.ID)
+            // Matches presets like "(A:TITLE,String)" in different variations. These will most likely be of type String an will therefore be treated as String.
+            Match stringPreset = Regex.Match(SimVarName, "\\(.*A.*:.*,.*String.*\\)");
```

Comment:
Valid presets are also (A:TITLE,string) and (A:TITLE,STRING), which also should be recognized.
So first convert the string .ToUpper() and then check for STRING in the regex or use a regex where case sensitivity does not matter.